src/test/java/com/jnape/palatable/lambdakoans/AboutApplicatives.java

```diff
@@ -0,0 +1,157 @@
+    }
+
+    @Test
+    public void lazyApplicatives() {
+        // Zipping LambdaIterables is lazy because LambdaIterables are lazy
+        LambdaIterable<Integer> infiniteOnes = LambdaIterable.wrap(repeat(1));
+        Fn1<Integer, Integer> inc = x -> x + 1;
+
+        LambdaIterable<Fn1<? super Integer, ? extends Integer>> wrappedInc = LambdaIterable.wrap(asList(inc));
+        LambdaIterable<Integer> zippedOnes = infiniteOnes.zip(wrappedInc);
+        assertThat(take(3, zippedOnes.unwrap()), iterates(__()));
+
+        // We might lazily get a mapping function...
+        AtomicInteger computed = new AtomicInteger(0);
+
+        Fn1<Integer, Maybe<Fn1<? super Integer, ? extends String>>> expensiveWayToGetMaybeToString = fn1((Integer n) -> {
+            for (int i = 0; i < n; i++) {
+                computed.incrementAndGet();
+            }
+            return just(Object::toString);
+        });
+
+        Lazy<Maybe<Fn1<? super Integer, ? extends String>>> lazyGetToString = lazy(() ->
+                expensiveWayToGetMaybeToString.apply(100_000_000));
+
+        // ...then apply it with lazyZip
+        Maybe<Integer> nothing = nothing();
+        Lazy<Maybe<String>> lazyNothingToString = nothing.lazyZip(lazyGetToString);
+
+        assertThat(lazyNothingToString.value(), equalTo(__));
+        assertThat(computed.get(), equalTo(__));
+
+        // zip, however, we've eagerly generated a mapping function
+        Maybe<String> nothingToString = nothing.zip(expensiveWayToGetMaybeToString.apply(100_000));
+        assertThat(nothingToString, equalTo(__));
+        assertThat(computed.get(), equalTo(__));
+    }
+
+    @Test
+    public void functionIsApplicative() {
+        Fn1<String, Integer> strLen = String::length;
+        Fn1<String, String> toUpper = String::toUpperCase;
+
+        // Result of unary function calls are passed to mapping function as arguments
+        Fn1<String, Tuple2<Integer, String>> lengthAndUppercase = LiftA2.liftA2(Tuple2::tuple, strLen, toUpper);
+        assertThat(lengthAndUppercase.apply("hello world"), equalTo(tuple(11, "HELLO WORLD")));
+
+        Fn1<Integer, Integer> mod3 = i -> i % 3;
+        Fn1<Integer, Integer> div3 = i -> i / 3;
+
+        Fn1<Integer, String> showDivision = LiftA2.liftA2((divided, remainder) -> String.format("%d * 3 + %d", divided, remainder), div3, mod3);
+        assertThat(showDivision.apply(10), equalTo(__));
+
+
+        Fn1<String, Integer> findStart = s -> s.indexOf('j');
+        Fn1<String, Integer> findEnd = s -> s.indexOf(' ');
+        Fn3<String, Integer, Integer, String> cutString = String::substring;
+
+        Fn1<String, String> transformAndCut = LiftA3.liftA3(cutString, toUpper, findStart, findEnd);
+        assertThat(transformAndCut.apply("hellojava world"), equalTo(__));
+    }
+
+    @Test
+    public void applicativeRepresentsParallelism() throws ExecutionException, InterruptedException {
```

Comment:
I wonder if we can write this test to timeout at a certain limit and the problem can be to change the executor to have enough threads to run them in parallel and make the test pass.